### ISSUE TITLE
Price range input placeholder/max values do not handle comma separated currencies (e.g. Euros)

### ISSUE THREAD

**Issue by willbroderick** [Shopify/dawn#255]
**Describe the current behavior**
Store setup:
Use Euros, set money format to "€{{amount_with_comma_separator}}", on a store with products containing prices in the thousands.
The price range filter inputs do not contain correct values.

See max/placeholder here:
![image](https://user-images.githubusercontent.com/1590737/126745571-16237223-88e5-4ad0-bc93-889c48d0fbfe.png)

Referring to the code here: https://github.com/Shopify/dawn/blob/main/sections/main-collection-product-grid.liquid#L147
`placeholder="{{ filter.range_max | money_without_currency | replace: ',', '' }}"`
This goes: 2836.10 -> "2.836,10" -> "2.83610"

**Describe the expected behavior**
The output should be `placeholder="2836.10"`

**Version information (Dawn, browsers and operating systems)**
- Dawn Version: 2.0.0-alpha.23
- Any OS/Browser

**Possible solution**
The ideal solution would be for Shopify to handle this conversion, perhaps like this:
`placeholder="{{ filter.range_max | money_to_input_value }}"`

**Additional context/screenshots**
In my opinion the logic is far too complex for theme code. I don't think it's reasonable to expect a theme developer to be aware of the nuances of all currencies and their delimiters, whether or not they have 0/2/3 decimal places, to know what is going to be output by the current shop.money_format formatting, and when to check against cart.currency.iso_code instead.

All of these appear to factor into this conversion, and I've yet to see an example that does this correctly.

Locking this logic down to the the state of the world in July 2021 also seems like an odd choice to me.

**Comment by bakura10** [comment]
I agree. Shopify definitely need to give us a better way to format the pricing.

I don't think also that the decimal value should be exposed at all in the filter, because the increment is usually 1, and can make the prices extremely long.

For now I have been able to create integer values using this, but I have no idea if this will work constantly for all currencies:

```
{%- assign min_value = filter.min_value.value | default: 0.0 | divided_by: 100.0 -%}
{%- assign max_value = filter.max_value.value | default: filter.range_max | divided_by: 100.0 -%}
```

I think Shopify should expose additional variables:

* {{ filter.min_value.formatted_value }}, {{ filter.max_value.formatted_value }}, {{ filter.formatted_range_max }}: this would return a pre-formatted values and max range to be used in the selector, without relying on hard-coded currencies list
* {{ filter.step }}: return the most appropriate step for the "number" input based on the currency. For instance, it could return 1 for euro, but 100 for yen.

**Comment by bakura10** [comment]
In addition of those improvements, I came into another issue.

When building the input for the price filter, I use the {{cart.currency.symbol}} to prepend the currency symbol:

![image](https://user-images.githubusercontent.com/1198915/134906511-64e101cf-517b-4b8d-8ec7-73a4071c4999.png)

Unfortunately, for some languages like French, currency code should be after, so it would make more sense to build it like this:

![image](https://user-images.githubusercontent.com/1198915/134906696-329cda1b-1e7e-462b-b718-69004fbed6d8.png)

But this is impossible for us to know as it depends on the active locale, and we can't hardcode this at theme level.

I therefore suggest the following improvement by adding a new {{filter.currency_prefix}} and {{filter.currency_suffix}} drops. Shopify would automatically make one or the other blank depending on the active locale, which would allow us to build a locale-aware price filter.

**Comment by willbroderick** [comment]
Would like to mention https://github.com/Shopify/theme-updates/issues/164 here.
On a new store with CLP currency:
```
product.price | money => $12.000
product.price | money_without_currency => 12,000
```
This makes price formatting difficult to handle or predict. 'money' seems to magically select a currency-specific format, but 'money_without_currency' uses the format in the currency settings.

**Comment by itshaunter** [comment]
Hey team, we've received a new report of this behaviour found on the merchant's storefront. Currency is Hungarian Forints (HUF) and inspecting the price filter fields confirms the max values are being replaced with decimal point conversions. eg `32,790` listed as a max of `32.79`:

![image](https://github.com/Shopify/dawn/assets/39280881/244fec36-0a0d-4566-b6c2-f8977f0b8f28)

Interestingly the `Drawer` filter layout does not exhibit this same issue in the Theme Editor environment, currently suggesting the merchant tests this live as a potential workaround: https://screenshot.click/08-27-at0iu-gwiwd.mp4

[Internal](https://app.shopify.com/services/internal/shops/62152540326)

**Comment by CianCroke** [comment]
Hi team, have a merchant experiencing this issue when attempting to filter by price on their storefront collections. Currency is set to Chilean Pesos (CLP), and when attempting to enter a price option, the filter is defaulting to display a different minimum amount than what's manually entered. 

[Internal](https://app.shopify.com/services/internal/shops/55589830727)
[Zendesk](https://shopify.zendesk.com/agent/tickets/43619119)
[Screen recording example from merchant](https://shopify.click/4722-63020-64819-38013-33101.webm) - SA was able to replicate unexpected behaviour via Theme Editor also : [screen recording](https://shopify.click/51486-97344-9735-65276-15547.webm)

**Comment by NeevusPeevus** [comment]
Hey team, another ticket with this issue on Refresh 12 and replicated on other themes. Currency is set to COP Colombian Peso. When entering price option the filter can display a different amount and products are not appearing once filtered. 

[Internal](https://app.shopify.com/services/internal/shops/83628622119)
[Zendesk](https://shopify.zendesk.com/agent/tickets/43933747)
[Screen recording](https://screenshot.click/09-48-scyf5-mhup4.mp4)

**Comment by PaulNewton** [comment]
@milan3015809 et al , while this should be a liquid issue, I've noticed people are showing that they are only checking with **devtools** for the computed DOM.
Out of curiosity is this "5.790" or similar cases also being checked if it's actually rendered in either the HTML source (ctrl+u)
, or any returned HTML from the **section rendering** api (_either dig into devtools network responses, or directly navigate to the url for the section to be rendered for the feature_).

Meaning:
if only the computed DOM is being checked then some cases could be a completely different issue for different sites with different javascript OR liquid.
if the rendered source is checked then it should be liquid for any specific case.
